crashtracker-ffi/src/datatypes/mod.rs

```diff
@@ -13,7 +13,7 @@ pub fn option_from_char_slice(s: CharSlice) -> anyhow::Result<Option<String>> {
 /// A generic result type for when a crashtracking operation may fail,
 /// but there's nothing to return in the case of success.
 #[repr(C)]
-pub enum CrashtrackerResult {
+pub enum Result {
     Ok(
         /// Do not use the value of Ok. This value only exists to overcome
         /// Rust -> C code generation.
```

Comment:
Very likely for a separate PR as well, but it occurred to me that both `ddog_crasht_Result` and `ddog_prof_Profile_Result` are exactly the same `(dummy boolean, ddog_Error)` pair.

Furthermore, they already both use `ddog_Error`, so I'm thinking it may be relevant to introduce a shared `ddog_Result`.

Reply:
https://github.com/DataDog/libdatadog/issues/563

Reply:
https://github.com/DataDog/libdatadog/issues/564

Reply:
https://github.com/DataDog/libdatadog/issues/563